app/models/queue_it/queue.rb

```diff
@@ -27,14 +27,16 @@ def two_nodes?
       size == 2
     end
 
-    def get_next_by_with_queue_length_one(nodable_attribute, attribute_value)
-      head_node if head_node&.nodable.send(nodable_attribute) == attribute_value
+    def get_next_by_with_queue_length_one(nodable_attribute, attribute_value = nil)
+      exp = create_expression(attribute: nodable_attribute, value: attribute_value)
```

Comment:
not sure if your refactor does the same thing.

the previous implementation does a filter instead of a sort

Reply:
This is not sorting.